### PR TITLE
Make sure the mandatory checkpoint includes Canopy activation

### LINKED ISSUE #2131: Make sure the mandatory checkpoint includes the Canopy activation block
**Is your feature request related to a problem? Please describe.**

In general, the chain history for each block commits to all the blocks between:
* the `previous network upgrade activation + 1` and
* the previous block

So the `Canopy` activation block commits to all the blocks between `Heartwood activation + 1` and `Canopy activation - 1`.

~~If we change the mandatory checkpoint to `Canopy activation + 1`, we will be able to skip calculating all the heartwood chain history.~~

The `ChainVerifier` checkpoints all blocks up to and including the `Canopy` activation maximum height. Zebra chooses the next checkpoint _at or after_ the mandatory height.

So the actual checkpoint does not need to change on either network:
| Network | Canopy | Zebra Checkpoint |
| -------- | ------- | ------ |
| Mainnet | 1028500 | 1028849 |
| Testnet | 1046400 | 1046400 |

But we do need to update the finalized state assertion, the documentation, and the checkpoint tests.

**Describe the solution you'd like**

- [x] assert that the canopy activation block is a finalized checkpoint block
- [x] clarify the checkpoint test docs - they're already checking that canopy activation is a checkpoint block 
- [x] update Zebra's checkpoint docs to explicitly include Canopy activation 
- [ ] if the mandatory checkpoint appears in multiple places, turn it into a named constant (optional)

The hard-coded checkpoint lists extend a long way past Canopy, so they don't need to be updated.

**Additional context**

See #2091 for more details

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

We want to ~change the mandatory checkpoint from Canopy to `Canopy + 1 block`~ make sure the mandatory checkpoint includes Canopy activation.

Details in https://github.com/ZcashFoundation/zebra/issues/2131

## Solution

- [x] assert that the canopy activation block is a finalized checkpoint block
- [x] clarify the checkpoint test docs - they're already checking that canopy activation is a checkpoint block 
- [x] update Zebra's checkpoint docs to explicitly include Canopy activation 
- [ ] if the mandatory checkpoint appears in multiple places, turn it into a named constant (optional)

The code in this pull request has:
  - [x] Documentation Comments
  - [x] Unit Tests

## Review

@oxarbitrage should review @teor2345's changes.

## Related Issues

Close https://github.com/ZcashFoundation/zebra/issues/2131

### CODE REVIEW

**Comment by daira** [comment]
It might be clearer to say "Canopy + 1 block". For a moment I thought you meant NU5.

**Comment by oxarbitrage** [comment]
> It might be clearer to say "Canopy + 1 block". For a moment I thought you meant NU5.

Thanks, i think you are right. I changed all the text in https://github.com/ZcashFoundation/zebra/pull/2235/commits/970ca0ab0b9ac33c2cabe5d80e7e0fd0fe740169

File: zebra-consensus/src/checkpoint/list/tests.rs
Comment (teor2345) [inline zebra-consensus/src/checkpoint/list/tests.rs]:
```suggestion
        "Pre-Canopy blocks and the Canopy activation block must be verified by checkpoints"
```

File: zebra-consensus/src/config.rs
Comment (teor2345) [inline zebra-consensus/src/config.rs]:
```suggestion
    /// (Zebra always checkpoints on Canopy activation + 1 block.)
```

**Comment by teor2345** [comment]
> It might be clearer to say "Canopy + 1 block". For a moment I thought you meant NU5.

To be clear, it's just the Canopy activation block we want to checkpoint. The new history tree starts at `Canopy activation`, which provides the history root for the `Canopy activation + 1 block` header.

**Review by daira — APPROVED** [review]
ACK

**Review by oxarbitrage — COMMENTED** [review]
I can't approve as i opened the PR but looks good, will merge.

**Comment by oxarbitrage** [comment]
Actually it seems i don't have the admin merge option available anymore. Instead i added an auto merge after someone else approves.

**Review by teor2345 — APPROVED** [review]
@oxarbitrage and I are happy here.

**Pull request merged**
